flink-metrics/flink-metrics-statsd/src/main/java/org/apache/flink/metrics/statsd/StatsDReporter.java

```diff
@@ -240,12 +240,6 @@ public String filterCharacters(String input) {
 	}
 
 	private boolean numberIsNegative(Number input) {
-		try {
-			return new BigDecimal(input.toString()).compareTo(BigDecimal.ZERO) == -1;
-		} catch (Exception e) {
-			//not all Number's can be converted to a BigDecimal, such as Infinity or NaN
-			//in this case we just say it isn't a negative number
-			return false;
-		}
+		return input.doubleValue() < 0;
```

Comment:
Is there a reason you went for the simple comparison instead of the suggested `Double#compareTo`? The javadocs list a number of special cases that are handled differently:

```
     * Compares two {@code Double} objects numerically.  There
     * are two ways in which comparisons performed by this method
     * differ from those performed by the Java language numerical
     * comparison operators ({@code <, <=, ==, >=, >})
     * when applied to primitive {@code double} values:
     *          {@code Double.NaN} is considered by this method 
     *               to be equal to itself and greater than all other
     *               {@code double} values (including
     *               {@code Double.POSITIVE_INFINITY}).
     *          {@code 0.0d} is considered by this method to be greater
     *               than {@code -0.0d}.
```

Reply:
Ah sorry, no reason! 

I've changed it to Double.compare(), I've also changed line #190 to use this method so the double check there works in the same way.